pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -191,7 +191,7 @@ func (srv PrometheusSrv) toRuleGroup(groupName string, folder *models.Folder, ru
 		Name: groupName,
 		File: folder.Title, // file is what Prometheus uses for provisioning, we replace it with namespace.
 	}
-
+	ngmodels.RulesGroup(rules).SortByGroupIndex()
```

Comment:
Does this change the order of the original slice `rules`? I thought `ngmodels.RuleGroup(rules)` would not change `rules` but instead create a new slice with the same pointers, however since we don't assign the new slice to a variable this is equivalent to a no-op. Perhaps I am wrong?

Reply:
I remember it does because while the new slice will have a new header (head, length and capacity), both slices will share the same data so sorting one of the slices will also sort the other!

Reply:
Yes, it does change the original slice. Although I prefer immutability too, the Go approach seems to be fine with mutating the original slice (e.g. sort.Slice).